### PR TITLE
Fixed wishlist issue for out of stock item

### PR DESCRIPTION
**Pull request by saphaljha**
### Fixed Issues (if relevant)

1. magento/magento2: Fixed wishlist issue for out of stock item

### Manual testing scenarios (*)

1. Add out of stock item into the wishlist

### Expected result (*)

expected

### Actual result (*)

actual

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hi @saphaljha,
thank you for your contribution!

There is an integration test failure:
```
1) Magento\Wishlist\Model\WishlistTest::testGetItemCollectionWithDisabledProduct
Failed asserting that an array is empty.
/home/travis/build/magento/magento2/dev/tests/integration/testsuite/Magento/Wishlist/Model/WishlistTest.php:122
```
Could you please take a look?

Please, also squash all commits into a single commit in order to cleanup the history.

Thank you!

**Comment by saphaljha** [comment]
Hi @dmytro-ch,

I have checked it and then I decided to remove that code, please have a look.

File: app/code/Magento/Wishlist/Model/ResourceModel/Item/Collection.php [hunk 1]
```diff
@@ -320,8 +319,6 @@ protected function _assignProducts()
                     $item->setName($product->getName());
                     $item->setPrice($product->getPrice());
                 }
-            } else {
```
Comment (dmytro-ch) [inline app/code/Magento/Wishlist/Model/ResourceModel/Item/Collection.php:323]: I don't think we need to remove this condition.
Most of the tests failures seem to be caused by this change.
E.g.
```
1) Magento\GraphQl\Wishlist\WishlistTest::testGetCustomerWishlist
Magento\Framework\Exception\LocalizedException: Cannot specify product.
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hi @saphaljha,
currently, the failing tests indicate that there is some issue in the new logic.
Please make sure the changes don't lead to additional issues for some specific cases by checking the failed tests.
Feel free to update the tests that are based on a wrong behavior of the wishlist feature.

Thank you!

File: app/code/Magento/Wishlist/Model/ResourceModel/Item/Collection.php (hunk 1 above)
Comment (saphaljha) [reply app/code/Magento/Wishlist/Model/ResourceModel/Item/Collection.php:323]: Hello @dmytro-ch , I'll check and update the code soon.

**Comment by engcom-Foxtrot** [comment]
@saphaljha issue you described is not reproducible. All works fine. We are closing this PR.

**Pull request closed without merging**